rust/operator-binary/src/druid_connection_controller.rs

```diff
@@ -0,0 +1,316 @@
+    },
+    SupersetDBExistsCheck {
+        source: stackable_operator::error::Error,
+    },
+    SupersetDBRetrieval {
+        source: stackable_operator::error::Error,
+    },
+}
+type Result<T, E = Error> = std::result::Result<T, E>;
+
+pub async fn reconcile_druid_connection(
+    druid_connection: DruidConnection,
+    ctx: Context<Ctx>,
+) -> Result<ReconcilerAction> {
+    tracing::info!("Starting reconciling DruidConnections");
+
+    let client = &ctx.get_ref().client;
+
+    if let Some(ref s) = druid_connection.status {
+        match s.condition {
+            DruidConnectionStatusCondition::Provisioned => {
+                // Is the superset DB object there, and is its status "Ready"?
+                let superset_db_ready = if client
+                    .exists::<SupersetDB>(
+                        &druid_connection.spec.superset_cluster_name,
+                        Some(&druid_connection.spec.superset_cluster_namespace),
+                    )
+                    .await
+                    .context(SupersetDBExistsCheck)?
+                {
+                    let superset_db_status = client
+                        .get::<SupersetDB>(
+                            &druid_connection.spec.superset_cluster_name,
+                            Some(&druid_connection.spec.superset_cluster_namespace),
+                        )
+                        .await
+                        .context(SupersetDBRetrieval)?
+                        .status;
+                    if let Some(s) = superset_db_status {
+                        s.condition == SupersetDBStatusCondition::Ready
+                    } else {
+                        false
+                    }
+                } else {
+                    false
+                };
+                // Is the referenced druid discovery configmap there?
+                let druid_discovery_cm_exists = client
+                    .exists::<ConfigMap>(
+                        &druid_connection.spec.druid_cluster_name,
+                        Some(&druid_connection.spec.druid_cluster_namespace),
+                    )
+                    .await
+                    .context(DruidDiscoveryCheck)?;
+
+                if superset_db_ready && druid_discovery_cm_exists {
+                    let superset_db = client
+                        .get::<SupersetDB>(
+                            &druid_connection.spec.superset_cluster_name,
+                            Some(&druid_connection.spec.superset_cluster_namespace),
+                        )
+                        .await
+                        .context(SupersetDBRetrieval)?;
+                    // Everything is there, retrieve all necessary info and start the job
+                    let sqlalchemy_str = get_sqlalchemy_uri_for_druid_cluster(
+                        &druid_connection.spec.druid_cluster_name,
+                        &druid_connection.spec.druid_cluster_namespace,
```

Comment:
These are reoccuring 5 times. I would extract that into an extra variable or offer a method for that within `DruidConnection`

Reply:
I did my best to use ObjectRefs now